pandas/core/arrays/timedeltas.py

```diff
@@ -395,6 +403,163 @@ def f(x):
 # ---------------------------------------------------------------------
 # Constructor Helpers
 
+def sequence_to_td64ns(data, copy=False, unit='ns', errors='raise'):
+    """
+    Parameters
+    ----------
+    array : list-like
+    copy : bool, default False
+    unit : str, default "ns"
+    errors : {"raise", "coerce", "ignore"}, default "raise"
+
+    Returns
+    -------
+    ndarray[timedelta64[ns]]
+    inferred_freq : Tick or None
+
+    Raises
+    ------
+    ValueError : data cannot be converted to timedelta64[ns]
+
+    Notes
+    -----
+    Unlike `pandas.to_timedelta`, if setting `errors=ignore` will not cause
+    errors to be ignored; they are caught and subsequently ignored at a
+    higher level.
+    """
+    inferred_freq = None
+    unit = parse_timedelta_unit(unit)
+
+    # Unwrap whatever we have into a np.ndarray
+    if not hasattr(data, 'dtype'):
+        # e.g. list, tuple
+        if np.ndim(data) == 0:
+            # i.e. generator
+            data = list(data)
+        data = np.array(data, copy=False)
+    elif isinstance(data, ABCSeries):
+        data = data._values
```

Comment:
Could potentially just get the freq here for Series[timedelta64[ns]]. If `TimedeltaArray` were done, we could change the following `elif` to an `if` and re-use that. For now I think we could check for timedelta dtype and do `inferred_freq = data.dt.freq` (which I assume is cheaper than re-inferring the freq?)

Reply:
Yah, I'm looking forward to this.

> and do `inferred_freq = data.dt.freq`

One of the first things I tried to do around here was get DatetimeIndex/TimedeltaIndex to have a `.dt` dummy property returning `self` so we wouldn't have to check for Index vs. Series; gotta get around to making that happen.